CHANGELOG.md

```diff
@@ -4,6 +4,17 @@ Changelog is rather internal in nature. See release notes for the public overvie
 
 ## Version 3.x.x (`release-v3` branch)
 
+- [#612]
+  - **Description:** KListWithOverflow fixes
```

Comment:
Before merging, could you please add a bit more of specificity? This will be used as a basis for release notes where we will mention what fixes are included in the release, from a consumer point of view.